docs/source/repository_structure.mdx

```diff
@@ -20,11 +20,50 @@ my_dataset_repository/
 └── test.csv
 ```
 
-## Splits and file names
+## Split pattern hierarchy
 
-🤗 Datasets automatically infer a dataset's train, validation, and test splits from the file names.
+🤗 Datasets searches for certain patterns in the dataset repository to automatically infer the dataset splits. There is an order to the patterns, beginning with the custom filename split format to treating all files as a single split if no pattern is found.
 
-All the files that contain a split name in their names (delimited by non-word characters, see below) are considered part of that split:
+### Custom filename split
+
+The most flexible pattern to accomodate other split types in addition to the traditional train, validation, and test sets, is to name your data files like `data/<split_name>-xxxxx-of-xxxxx.csv`.
```

Comment:
```suggestion
If your dataset has split that have non traditional names, you should name your data files like `data/<split_name>-xxxxx-of-xxxxx.csv`.
```
Not the best wording in my suggestion, feel free to change, I just wanted to point out that I wouldn't encourage for this pattern, I mean no need to name it the most flexible. it's quite a specific one. Putting all the data into a single directory named as a split (the next pattern) is easier and for most cases enough.